cranelift/codegen/meta/src/isa/x86.rs

```diff
@@ -174,38 +174,200 @@ fn define_settings(shared: &SettingGroup) -> SettingGroup {
+        "ivybridge",
+        "Ivy Bridge microarchitecture.",
+        preset!(sandy_bridge),
+    );
+    settings.add_preset(
+        "core-avx-i",
+        "Intel Core CPU with 64-bit extensions.",
+        preset!(ivy_bridge),
+    );
     let haswell = settings.add_preset(
         "haswell",
         "Haswell microarchitecture.",
         preset!(nehalem && has_bmi1 && has_bmi2 && has_lzcnt),
     );
+    settings.add_preset(
+        "core-avx2",
+        "Intel Core CPU with AVX2 extensions.",
+        preset!(haswell),
+    );
     let broadwell = settings.add_preset(
         "broadwell",
         "Broadwell microarchitecture.",
         preset!(haswell && has_fma),
     );
     let skylake = settings.add_preset("skylake", "Skylake microarchitecture.", preset!(broadwell));
+    let knights_landing = settings.add_preset(
+        "knl",
+        "Knights Landing microarchitecture.",
+        preset!(has_popcnt && has_avx512f && has_fma && has_bmi1 && has_bmi2 && has_lzcnt),
+    );
+    settings.add_preset(
+        "knm",
+        "Knights Mill microarchitecture.",
+        preset!(knights_landing),
+    );
+    let skylake_avx512 = settings.add_preset(
+        "skylake-avx512",
+        "Skylake AVX512 microarchitecture.",
+        preset!(broadwell && has_avx512f && has_avx512dq && has_avx512vl),
+    );
+    settings.add_preset(
+        "skx",
+        "Skylake AVX512 microarchitecture.",
+        preset!(skylake_avx512),
+    );
+    let cascadelake = settings.add_preset(
+        "cascadelake",
+        "Cascade Lake microarchitecture.",
+        preset!(skylake_avx512),
+    );
+    settings.add_preset(
+        "cooperlake",
+        "Cooper Lake mircoarchitecture.",
+        preset!(cascadelake),
+    );
     let cannonlake = settings.add_preset(
         "cannonlake",
         "Canon Lake microarchitecture.",
-        preset!(skylake),
+        preset!(skylake && has_avx512f && has_avx512dq && has_avx512vl && has_avx512vbmi),
     );
-    settings.add_preset(
+    let icelake = settings.add_preset(
         "icelake",
         "Ice Lake microarchitecture.",
-        preset!(cannonlake),
+        preset!(cannonlake && has_avx512bitalg),
+    );
+    settings.add_preset(
+        "icelake-client",
+        "Ice Lake microarchitecture.",
+        preset!(cannonlake && has_avx512bitalg),
     );
+    let icelake_server = settings.add_preset(
+        "icelake-server",
+        "Ice Lake (server) microarchitecture.",
+        preset!(icelake),
+    );
+    settings.add_preset(
+        "tigerlake",
+        "Tiger Lake microarchitecture.",
+        preset!(icelake),
+    );
+    let sapphire_rapids = settings.add_preset(
+        "sapphirerapids",
+        "Saphire Rapids microarchitecture.",
+        preset!(icelake_server),
+    );
+    let alderlake = settings.add_preset(
+        "alderlake",
+        "Alder Lake microarchitecture.",
+        preset!(tremont && has_fma && has_bmi1 && has_bmi2 && has_lzcnt),
```

Comment:
Is this duplicated from up above?

Reply:
I think the duplicate definition of `alderlake` is the last issue remaining here. Once that's fixed I think this PR is ready!